internal/verifier/verifier_test.go

```diff
@@ -34,11 +37,47 @@ func TestReproducibleProvenanceVerifier(t *testing.T) {
 	testutil.Chdir(t, "../../")
 	verifier := ReproducibleProvenanceVerifier{}
 
-	if err := verifier.Verify(schemaExamplePath); err != nil {
+	if err := verifier.Verify(validProvenancePath); err != nil {
 		t.Fatalf("couldn't verify the provenance file: %v", err)
 	}
 }
 
+func TestReproducibleProvenanceVerifier_invalidHash(t *testing.T) {
+	// The path to provenance is specified relative to the root of the repo, so we need to go one level up.
+	// Get the current directory before that to restore the path at the end of the test.
+	currentDir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("couldn't get current directory: %v", err)
+	}
+	defer testutil.Chdir(t, currentDir)
+	testutil.Chdir(t, "../../")
+	verifier := ReproducibleProvenanceVerifier{}
+
+	want := "failed to verify the hash of the built binary"
+
+	if err := verifier.Verify(invalidHashProvenancePath); !strings.Contains(err.Error(), want) {
+		t.Fatalf("want error containing message %q, got %v", want, err)
+	}
+}
+
+func TestReproducibleProvenanceVerifier_badCommand(t *testing.T) {
```

Comment:
Really cool test! As idea: If we need the `bad_command_provenance.json` only in this test, shall we maybe hard code it as string in the test? Might also be faster to see the whole test then.

Similar with the `invalidHashProvenancePath`?